pennylane/labs/resource_estimation/resource_container.py

```diff
@@ -337,3 +334,10 @@ def _scale_dict(dict1: defaultdict, scalar: int, in_place=False):
         combined_dict[k] *= scalar
 
     return combined_dict
+
+
+def _make_hashable(d: dict) -> tuple:
+    if isinstance(d, dict):
+        return tuple((name, _make_hashable(value)) for name, value in d.items())
+
+    return d
```

Comment:
Nice! I like this idea. Just going to remove the type hint since it's not true in general. 
```suggestion
def _make_hashable(d) -> tuple:
    if isinstance(d, dict):
        return tuple((name, _make_hashable(value)) for name, value in d.items())
    return d
```